code/game/objects/structures/shower.dm

```diff
@@ -23,12 +27,16 @@
 	///What reagent should the shower be filled with when initially built.
 	var/reagent_id = /datum/reagent/water
 	///How much reagent capacity should the shower begin with when built.
-	var/reaction_volume = 200
+	var/reagent_capacity = 200
+	///How many units the shower refills every second.
+	var/refill_rate = 0.5
+	/// Whether the shower can refill.
+	var/can_refill = (SHOWER_CAN_REFILL|SHOWER_CAN_TOGGLE_REFILL)
```

Comment:
I'm not sure if the performance saved by having one variable rather than two, is worth the complexity hit?

Reply:
Neither am I, but last I heard we were running low on memory.

Reply:
we aren't "THERE'S TOO MANY BOOLS, SHUT IT DOWN" running low on memory, we're "I swear to god stop making miniature memory leaks" low on memory. Up to you

Reply:
Made it a pair of bools. It's not as compact, but it works just as well.